sdk/messaging/azservicebus/messageSettler.go

```diff
@@ -62,13 +53,22 @@ type CompleteMessageOptions struct {
 }
 
 // CompleteMessage completes a message, deleting it from the queue or subscription.
-func (s *messageSettler) CompleteMessage(ctx context.Context, message *ReceivedMessage, options *CompleteMessageOptions) error {
-	return s.settleWithRetries(ctx, message, func(receiver amqpwrap.AMQPReceiver, rpcLink amqpwrap.RPCLink) error {
-		if s.useManagementLink(message, receiver) {
-			return internal.SendDisposition(ctx, rpcLink, receiver.LinkName(), bytesToAMQPUUID(message.LockToken), internal.Disposition{Status: internal.CompletedDisposition}, nil)
-		} else {
-			return receiver.AcceptMessage(ctx, message.RawAMQPMessage.inner)
+func (ms *messageSettler) CompleteMessage(ctx context.Context, message *ReceivedMessage, options *CompleteMessageOptions) error {
+	return ms.settleWithRetries(ctx, message, func(receiver amqpwrap.AMQPReceiver, rpcLink amqpwrap.RPCLink) error {
+		var err error
+
+		if shouldSettleOnReceiver(message) {
+			ms.notifySettleOnLink(message)
+			err = receiver.AcceptMessage(ctx, message.RawAMQPMessage.inner)
 		}
+
+		if shouldSettleOnMgmtLink(err, message) {
```

Comment:
Does this intentionally fall through from the previous clause? If so, might be good to comment why.

Reply:
I added a comment in all four locations.